src/verify_permutation.rs

```diff
@@ -265,6 +265,16 @@ mod tests {
         assert!(gadget.execute(&mut WireValues::new()));
     }
 
+    #[test]
+    fn route_4x4() {
+        let mut builder = GadgetBuilder::<F257>::new();
+        builder.assert_permutation(
+            &[1u8.into(), 2u8.into(), 3u8.into(), 4u8.into()],
+            &[2u8.into(), 1u8.into(), 4u8.into(), 3u8.into()]);
+        let gadget = builder.build();
+        assert!(gadget.execute(&mut WireValues::new()));
+    }
+
```

Comment:
Added 4x4 test to cover even-numbered child switch branch.

Reply:
Good idea